frontend/model/contracts/chatroom.js

```diff
@@ -367,11 +371,10 @@ sbp('chelonia/defineContract', {
         const mentions = makeMentionFromUsername(me)
         const isIncludeMention = data.text.includes(mentions.me) || data.text.includes(mentions.all)
         if (!isAlreadyAdded && isIncludeMention) {
-          // TODO: Not sure createdDate should be this way
           addMention({
             contractID,
             messageId: data.id,
-            datetime: meta.createdDate,
+            datetime: data.createdDate, // TODO: Not sure createdDate should be this way
```

Comment:
Can you explain why you are using `data.createdDate` here instead of `meta.createdDate`? What does this represent?

Reply:
That's because this `datetime` is used to decide if the this mention should be removed or not.
Let's say the last message was a mention.
If user reads the last message, the mention is removed from the list of mentions.
And also the `UnreadSince.messageId` is the ID of the last message which made mention.
Here, if user edits the message(the last one) for example, user can update mention from @all to @alex or just update original message content, the mention is created again.
If the new mention's datetime is the editing time, it is later than the UnreadSince.createdDate.
So we can not delete this mention unless user creates a new message and read it.
That's why I used the `data.createdDate` instead of `meta.createdDate`.

Reply:
OK, I'm having difficulty following, and I don't think you answered what `data.createdDate` represents. What time does it represent? What time is stored in it?

Reply:
`meta.createdDate` : The date when edit this message
`data.createdDate` : The date when the message was created

Reply:
>  data.createdDate : The date when the message was created

OK, I see.

Normally that information is stored on the message object itself when it was created via `'gi.contracts/chatroom/addMessage'`.

Is there a reason you can't do that here and have to instead pass it in as part of the message edit?

Reply:
Correct, we can not do this in `process` or `sideEffect` or addMessage.
This happens only when the message is edited.

Reply:
Right, but I asked if there was a reason that it has to be passed in here like this, instead of being stored when `addMessage` is called?

Can you add in a comment and explain the reason?

Reply:
Okay.